### PR TITLE
Add converters - to be used in ESDT transfer construction (and parsing)

### PR DESCRIPTION
**Pull request by andreibancioiu**
Add a few converters, to be used for creating (and parsing) ESDT transfers - in a following PR.

### CODE REVIEW

File: server/services/converters.go [hunk 1]
```diff
@@ -70,3 +71,39 @@ func indexToOperationIdentifier(index int) *types.OperationIdentifier {
 func timestampInMilliseconds(timestamp int64) int64 {
 	return timestamp * 1000
 }
+
+func utf8ToHex(value string) string {
+	encoded := hex.EncodeToString([]byte(value))
```
Comment (mariusmihaic) [inline server/services/converters.go:76]: [comment]
You don't actually check here if the string is utf

File: server/services/converters.go [hunk 2]
```diff
@@ -70,3 +71,39 @@ func indexToOperationIdentifier(index int) *types.OperationIdentifier {
...
+		return "0" + hexString
+	}
+
+	return hexString
+}
+
+func amountToHex(value string) string {
+	bigAmount, ok := big.NewInt(0).SetString(value, 10)
+	if !ok {
+		return ""
```
Comment (mariusmihaic) [inline server/services/converters.go:92]: Why not return error?

File: server/services/converters.go [hunk 3]
```diff
@@ -70,3 +71,39 @@ func indexToOperationIdentifier(index int) *types.OperationIdentifier {
...
+
+func hexToAmount(hexString string) (string, error) {
+	amountBytes, err := hex.DecodeString(hexString)
+	if err != nil {
+		return "", err
+	}
+
+	amountBig := big.NewInt(0).SetBytes(amountBytes)
+	amount := amountBig.String()
+	return amount, nil
```
Comment (mariusmihaic) [inline server/services/converters.go:108]:
```suggestion
	return amountBig.String(), nil
```

File: server/services/converters_test.go [hunk 4]
```diff
@@ -33,3 +33,33 @@ func TestBlockIdentifierToAccountQueryOptions(t *testing.T) {
...
 	})
 }
+
+func TestUtf8ToHex(t *testing.T) {
+	require.Equal(t, "68656c6c6f", utf8ToHex("hello"))
+	require.Equal(t, "776f726c64", utf8ToHex("world"))
+}
+
+func TestAmountToHex(t *testing.T) {
+	require.Equal(t, "", amountToHex("0"))
```
Comment (mariusmihaic) [inline server/services/converters_test.go:43]: Is this really ok?

**Review by iulianpascalau — APPROVED** [review]

File: server/services/converters_test.go [hunk 5]
```diff
@@ -33,3 +33,33 @@ func TestBlockIdentifierToAccountQueryOptions(t *testing.T) {
 		require.ErrorContains(t, err, "encoding/hex: invalid byte")
 	})
 }
+
+func TestUtf8ToHex(t *testing.T) {
+	require.Equal(t, "68656c6c6f", utf8ToHex("hello"))
+	require.Equal(t, "776f726c64", utf8ToHex("world"))
```
Comment (BeniaminDrasovean) [inline server/services/converters_test.go:39]: Add a test with an odd num of hex chars.

File: server/services/converters_test.go (hunk 4 above)
Comment (andreibancioiu) [reply server/services/converters_test.go:43]: Yes, in our context, on most purposes, `""` is equivalent to `0`.

File: server/services/converters.go (hunk 3 above)
Comment (andreibancioiu) [reply server/services/converters.go:108]: Thanks, fixed.

File: server/services/converters.go (hunk 1 above)
Comment (andreibancioiu) [reply server/services/converters.go:76]: Renamed to `stringToHex`.

File: server/services/converters_test.go (hunk 5 above)
Comment (andreibancioiu) [reply server/services/converters_test.go:39]: Added.

**Review by BeniaminDrasovean — APPROVED** [review]

File: server/services/converters.go (hunk 2 above)
Comment (andreibancioiu) [reply server/services/converters.go:93]: Logging an error, thanks!

**Review by mariusmihaic — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**